### PR TITLE
Set true values in API for Safari with version_added: null

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
**Reviews and suggestions are highly encouraged and welcomed, especially as this is an almost entirely automated process, which hardly takes runtime flags into account.  See something that looks inaccurate?  Let me know!**

This is to help the efforts of 2019 KR by replacing the `null` values with boolean values where applicable.  All true values were identified within an IDL file in the WebKit source.  This does not replace any values set to a version number.  This PR is intended to remove as many null values as possible, and lead towards producing "real" values down the line.

I wrote the following JavaScript and Python files to create a list of all the values set to null and check upon their implementation in the IDL files respectively: https://gist.github.com/vinyldarkscratch/db303f7e72a582d2bf89d84bc4d035e2

Stats before and after this PR (based upon v0.0.73):

| browser | real values | `true` values | `null` values |
| --- | --- | --- | --- |
| total (before) | 55.87% | 22.49% | 21.64% |
| total (after) | 55.91% (+0.04%) | 25.19% (+2.70%) | 18.90% (-2.74%) |
| safari (before) | 49.77% | 19.16% | 31.07% |
| safari (after) | 49.93% (+0.16%) | 30.55% (+11.39%) | 19.52% (-11.55%) |
| safari ios (before) | 41.59% | 20.64% | 37.79% |
| safari ios (after) | 41.74% (+0.15%) | 30.89% (+10.25%) | 27.37% (-10.42%) |

### CODE REVIEW

**Comment by foolip** [comment]
As with https://github.com/mdn/browser-compat-data/pull/3658 there will be errors here of at least these kinds:
- things in WebKit's IDL which haven't shipped yet in Safari
- things which are shipped in Safari but where the Web IDL in WebKit is named differently
- interface mixins

**Comment by vinyldarkscratch** [comment]
I don’t doubt it, there’s sure to be errors in precise implementation in Safari!  I realized that, while seeing your comments on the other PRs, I had the latest version of the WebKit source, not the particular version that was deployed for Safari 12.  I knew, however, that there would be certain instances that would have to be checked by hand — as such, I didn’t create any “false” values unless I was absolutely certain they were false.  That leaves only potential for false positives, luckily (which in my opinion, is far better to have than false negatives).

I’m always looking for review/a second pair of eyes.  Any and all feedback is highly appreciated!

**Comment by foolip** [comment]
Ah, can you update the description to say this is only setting true values?

Do you think it'd make sense to first create a PR with null&rarr;true that have been confirmed by running tests? That's hopefully a lot, and makes it easier to review what remains since those are more likely to be incorrect.

**Comment by foolip** [comment]
@vinyldarkscratch since I already had the data and scripts I sent https://github.com/mdn/browser-compat-data/pull/3700 and humbly request your review of that. I'm going to try rebasing this PR on top of that to see what kinds of changes will remain to review.

File: api/AbsoluteOrientationSensor.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
             "version_added": "56"
           },
           "safari": {
-            "version_added": null
+            "version_added": true
```
Comment (foolip) [inline api/AbsoluteOrientationSensor.json]: This is the first change that remains after rebasing on top of https://github.com/mdn/browser-compat-data/pull/3700. AFAICT this IDL file isn't in WebKit at all, there's just LayoutTests/imported/w3c/web-platform-tests/interfaces/orientation-sensor.idl which is for testing.

I think you'll need to verify the paths you ingest and regenerate this PR.

Comment (vinyldarkscratch) [reply api/AbsoluteOrientationSensor.json]: Can do, yeah!

**Comment by Elchi3** [comment]
How much overlap is there between this PR and https://github.com/mdn/browser-compat-data/pull/3700? I guess if both PRs update a subset from null to true we could trust these with relatively high confidence.

**Comment by vinyldarkscratch** [comment]
It looks like #3700 doesn't have anything extra that this PR doesn't already have, and there appears to be 1575 changes across 244 files that are in this PR but not the other, primarily regarding the ANGLE/EXT/OES/WebGL, SVG, and RTC APIs.  So, we can trust the accuracy of #3700.

**Comment by foolip** [comment]
I've also confirmed that https://github.com/mdn/browser-compat-data/pull/3700 is a perfect subset of this by  reverting the commit from that PR on top of this PRs branch. That would have resulted in conflicts if https://github.com/mdn/browser-compat-data/pull/3700 contained changes that this does not.

Doing that leaves only 181 files with changes in this PR, which will be less to deal with.

**Comment by Elchi3** [comment]
Okay, given this, I think we can merge #3700. 
We have two independent approaches and I think that is enough for approving null -> true changes.

**Comment by Elchi3** [comment]
Merged #3700 and there are indeed no merge conflicts. :+1: 

Can you please rebase this PR, so the changeset becomes smaller?

**Comment by vinyldarkscratch** [comment]
Rebase complete!

File: api/AnimationEffect.json [hunk 2]
```diff
@@ -46,10 +46,10 @@
             "version_added": null
           },
           "safari": {
-            "version_added": null
+            "version_added": true
```
Comment (foolip) [inline api/AnimationEffect.json]: This is disconfirmed by evaluating `AnimationEffect` in Safari 12, it's not exposed. It is in Safari Technology Preview, so maybe the next version of Safari.

The cause is probably a failure to take into account `EnabledAtRuntime=WebAnimations`:
https://github.com/WebKit/webkit/blob/071e0ad875b4db3087abd4d88efafe83fdd78fb9/Source/WebCore/animation/AnimationEffect.idl#L27

**Comment by Elchi3** [comment]
This has been open for a long time now and basically the reason why we haven't merged it is still as described in https://github.com/mdn/browser-compat-data/pull/3658#issuecomment-484427362

I'm closing this as it is unlikely to merged currently. It can be re-opened if we gain new confidence with this approach or want to experiment with it again.

**Pull request closed without merging**